### ISSUE TITLE
Does not seem to support --arg flag

### ISSUE THREAD

**Issue by duro** [kislyuk/yq#16]
Been trying the following:

```
cat file.yml | yq --arg foo bar '.metadata.name = $foo'
```

And I get the following error:

```
yq: error: argument files: can't open 'bar': [Errno 2] No such file or directory: 'bar'
```

It would seem that jq's expected syntax for setting args with a space between key and value is tripping things up.

**Comment by kislyuk** [comment]
Thanks for reporting. While I look into the arg parsing issue, you might look into passing environment variables (`'.metadata.name = env.foo'`) or using shell string substitution of environment variables as a workaround.

**Comment by kislyuk** [comment]
Thanks again for reporting. Fix released in v2.3.5, please test.
